client.go

```diff
@@ -6,54 +6,61 @@ import (
 	"net"
 	"os"
 	"strings"
+	"sync"
 )
 
 func main() {
-	// Create new client connction
+	// Create new client connection
 	conn, err := net.Dial("tcp", "127.0.0.1:1234")
 	if err != nil {
-		fmt.Errorf("Failed to dial to the address. Error: %s", err)
+		fmt.Fprintf(os.Stderr, "Failed to dial to the address. Error: %s\n", err)
+		os.Exit(1)
 	}
 
 	defer conn.Close()
 	fmt.Println("Chatting...")
 
-	done := make(chan bool, 1)
-	// Read message from user and sent to the other user
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	// Read message from user and send to the other user
 	go func() {
+		defer wg.Done()
 		reader := bufio.NewReader(os.Stdin)
 		for {
+			fmt.Print("-> ")
```

Comment:
This print and line 37 print are not clean during first message. Output is something like "-> <- hello"